### PR TITLE
Add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by vitaliy-guliy**
### What does this PR do?
Adds `CONTRIBUTING.md` file describing the development flow.
Updates `devfile.yaml`.
Adds formatter to `package.json`. 

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/14126

Still not working command running in container (need to found a solution):
- [ ] package binaries
- [ ] create a workspace

Before merging the link to `chectl` sources in the devfile will be changed to original git repo.

### CODE REVIEW

File: devfile.yaml [hunk 1]
```diff
@@ -33,17 +33,51 @@ components:
...
+
+  - name: Install Packager
+    actions:
+      - type: exec
+        component: dev
+        command: >
+                 npm install -g pkg
+        workdir: /projects/chectl
+
+  - name: Package Binaries
```
Comment (benoitf) [inline devfile.yaml]: just to know, it's the deprecated binaries

it's `oclif-dev pack` now

File: devfile.yaml [hunk 2]
```diff
@@ -33,17 +33,51 @@ components:
...
 
+  - name: Run
+    actions:
+      - type: exec
+        component: dev
+        command: >
+                 ./run --help
+        workdir: /projects/chectl/bin
+
+  - name: Install Packager
```
Comment (benoitf) [inline devfile.yaml]: not required anymore (only for deprecated binaries)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+- [build](#build)
+- [run tests](#run-tests)
+- [package binaries](#package-binaries)
+- [code](#code)
+- [push changes, provide pull request](#push-changes-provide-pull-request)
+
+## Create workspace, clone sources
+
+To create a workpace you can use [devfile](devfile.yaml).
+For that you have to [install the latest version](https://www.eclipse.org/che/docs/che-7/che-quick-starts.html#installing-the-chectl-management-tool) of this chectl and create a workspace:
```
Comment (benoitf) [inline CONTRIBUTING.md]: it looks odd to install chectl when we clone chectl ?

File: devfile.yaml (hunk 1 above)
Comment (vitaliy-guliy) [reply devfile.yaml]: Thanks. I will try

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+After starting the workspace Theia will clone sources of `chectl` to `/projects/chectl` directory. Sources will be accessible in `theia-dev` and `dev` containers.
+
+#### 'theia-dev' container
+
+This container is used only for running Theia editor and does not take part in the following flow.
+
+#### 'dev' container
+
+`chectl` is written on TypeScript. For that this container has preinstalled software to be able to build, test and launch `chectl`.
```
Comment (benoitf) [inline CONTRIBUTING.md]:
```suggestion
`chectl` is written in TypeScript. For that this container has preinstalled software to be able to build, test and launch `chectl`.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+You workspace is initialized with a list of commands described in the [devfile](devfile.yaml) in `commands` section. Those command allow you to:
+- build
+- test
+- run
+- install packager
+- package binaries
+- format sources
+
+You can run commands in one of tree ways.
```
Comment (benoitf) [inline CONTRIBUTING.md]:
```suggestion
You can run commands in one of three ways.
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+```bash
+./bin/run --help
+```
+
+To test ensure `chectl` is built successfully, launch `Run` command. It wil run `chectl` with `--help` directive.
+
+## Package binaries
+
+### Install packager
```
Comment (benoitf) [inline CONTRIBUTING.md]: needs to be reworked. It's only using now `oclif-dev pack`

File: CONTRIBUTING.md (hunk 4 above)
Comment (vitaliy-guliy) [reply CONTRIBUTING.md]: Thanks

File: CONTRIBUTING.md (hunk 3 above)
Comment (vitaliy-guliy) [reply CONTRIBUTING.md]: Yes and no. It's ok if we are explaining the user steps to create a workspace.
From other side it's extra information as user is already pretty experienced if he is going to contribute to chectl.

File: CONTRIBUTING.md (hunk 5 above)
Comment (vitaliy-guliy) [reply CONTRIBUTING.md]: Thanks

File: devfile.yaml (hunk 1 above)
Comment (vitaliy-guliy) [reply devfile.yaml]: `oclif-dev pack` does not create executable file. It creates a package, which contains `node` with a set of javascript files.

File: devfile.yaml (hunk 2 above)
Comment (vitaliy-guliy) [reply devfile.yaml]: I have removed this command and updated a command to build packaged.
I will update it when it become clean how to use `oclif-dev pack`.

File: CONTRIBUTING.md (hunk 3 above)
Comment (vitaliy-guliy) [reply CONTRIBUTING.md]: Extra sentence has been removed.

File: devfile.yaml (hunk 1 above)
Comment (vitaliy-guliy) [reply devfile.yaml]: Reworked

File: devfile.yaml (hunk 2 above)
Comment (vitaliy-guliy) [reply devfile.yaml]: Removed

File: CONTRIBUTING.md (hunk 6 above)
Comment (vitaliy-guliy) [reply CONTRIBUTING.md]: Reworked

**Comment by benoitf** [comment]
FYI @vitaliy-guliy the semantic checks are failing https://github.com/che-incubator/chectl#providing-pull-request

**Comment by vitaliy-guliy** [comment]
> FYI @vitaliy-guliy the semantic checks are failing https://github.com/che-incubator/chectl#providing-pull-request

I don't understand why. Each commit has signed-off part.

File: devfile.yaml [hunk 7]
```diff
@@ -1,28 +1,29 @@
...
+  # - name: chectl
+  #   source:
+  #     type: git
+  #     location: 'https://github.com/che-incubator/chectl.git'
+
   - name: chectl
     source:
       type: git
-      location: 'https://github.com/che-incubator/chectl.git'
+      location: 'https://github.com/vitaliy-guliy/chectl.git'
```
Comment (benoitf) [inline devfile.yaml]: will be reverted before merging ?

**Comment by benoitf** [comment]
@vitaliy-guliy semantic commits check is not Signed commits
https://conventionalcommits.org

File: devfile.yaml (hunk 7 above)
Comment (vitaliy-guliy) [reply devfile.yaml]: Reverted

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+$ chectl workspace:start -f https://raw.githubusercontent.com/che-incubator/chectl/master/devfile.yaml
+```
+
+> See more about [devfile](https://redhat-developer.github.io/devfile)
+
+After starting the workspace Theia will clone sources of `chectl` to `/projects/chectl` directory. Sources will be accessible in `theia-dev` and `dev` containers.
+
+#### 'theia-dev' container
+
+This container is used only for running Theia editor and does not take part in the following flow.
```
Comment (benoitf) [inline CONTRIBUTING.md]: I'm not sure to understand why we have a container that is not used ?
AFAIK we don't have theia-dev there

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+uri=$(cat /projects/clbin); ./run workspace:start -f=$uri
+```
+
+To create a workspsace run `Create Workspace` command from `My Workspace`. The command will upload content of the generated `/projects/sample.devfile` devfile to https://clbin.com. And then will use given public URI to the devfile when running `chectl`.
+
+> See more about [clbin](https://clbin.com/)
+
+## Package binaries
+
+For packaging binaries we use https://github.com/oclif/dev-cli. It generates packaged for Linux, Windows and MacOS operation systems and puts the result in `dist/channels/stable` directory.
```
Comment (benoitf) [inline CONTRIBUTING.md]:
```suggestion
For packaging binaries, [oclif](https://github.com/oclif/dev-cli) is used. It generates packages for Linux, Windows and MacOS operation systems and puts the result in `dist/channels/stable` directory.
```

File: CONTRIBUTING.md (hunk 8 above)
Comment (vitaliy-guliy) [reply CONTRIBUTING.md]: theia-de, not theia-dev. It's my wrong.

File: CONTRIBUTING.md (hunk 9 above)
Comment (vitaliy-guliy) [reply CONTRIBUTING.md]: Thanks

**Pull request merged**